src/libraries/System.Private.CoreLib/src/System/Threading/DeferredDisposableLifetime.cs

```diff
@@ -50,7 +50,8 @@ public bool AddRef()
                 int oldCount = Volatile.Read(ref _count);
 
                 // Have we been disposed?
-                ObjectDisposedException.ThrowIf(oldCount < 0, this);
+                if (oldCount < 0)
+                    throw new ObjectDisposedException(typeof(T).ToString());
```

Comment:
Thanks for fixing. We'll want to fix the analyzer/fixer as well to not fire in this case. But that's not pressing and I can do that when I'm back.

Reply:
> We'll want to fix the analyzer/fixer as well to not fire in this case. But that's not pressing and I can do that when I'm back.

https://github.com/dotnet/roslyn-analyzers/pull/6396